diff -uNr acpi.py-0.1.0-org/acpi.py/acpi.py acpi.py-0.1.0/acpi.py/acpi.py
--- acpi.py-0.1.0-org/acpi.py/acpi.py	Thu Apr 10 22:20:48 2003
+++ acpi.py-0.1.0/acpi.py/acpi.py	Wed Oct 15 12:30:59 2003
@@ -32,7 +32,8 @@
 
 
 import os
-
+import sys
+import stat
 
 #enums
 OFFLINE     =  0
@@ -163,9 +164,10 @@
 			line = info_file.readline()
 			
 			while len(line) != 0:
+				line = line.lower()
 				if line.find("design capacity") == 0:
 					cap = line.split(":")[1].strip()
-					self.design_capacity[i] = int(cap.split("mAh")[0].strip())
+					self.design_capacity[i] = int(cap.split("mah")[0].strip())
 					break
 					
 				line = info_file.readline()
@@ -173,20 +175,21 @@
 			
 			
 	def update(self):
-		"""Read /proc/acpi/battery/*/state and extract needed infos"""
+		"""Read /proc/acpi/battery/*/status and extract needed infos"""
 		
 		for i in self.batteries:
-			state_file = open(self.proc_battery_dir + "/" + i + "/state")
+			state_file = open(self.proc_battery_dir + "/" + i + "/status")
 			line = state_file.readline()
 			
 			while len(line) != 0:
+				line = line.lower()
 				if line.find("remaining capacity") == 0:
 					cap = line.split(":")[1].strip()
-					self.life_capacity[i] = int(cap.split("mAh")[0].strip())
+					self.life_capacity[i] = int(cap.split("mah")[0].strip())
 					
 				#a little bit tricky... if loading of ac driver fails, we cant use info
 				#from /proc/ac_*/...
-				#if information in /proc/acpi/battery/*/state is wrong we had to
+				#if information in /proc/acpi/battery/*/status is wrong we had to
 				#track the capacity history.
 				#I asume that all battery state files get the same state.
 				if line.find("charging state") == 0:
@@ -223,9 +226,11 @@
 
 
 	def nb_of_batteries(self):
-		#returns the number of batteries
-		#if it returns 0, maybe ACPI is not available or 
-		#battery driver is not load
+		"""
+		Returns the number of batteries
+		if it returns 0, maybe ACPI is not available or 
+		battery driver is not load
+		"""
 		return len(self.batteries)
 		
 
